lib/ovirt/service.rb

```diff
@@ -145,8 +155,40 @@ def status(link)
       node.xpath('status/state').text
     end
 
+    # Checks if the API is available in the given candidate path. It does so sending a request without
+    # authentication. If the API is available there it will respond with the "WWW-Autenticate" header
+    # and with the "RESTAPI" or "ENGINE" realm.
+    def probe_api_path(uri, path)
+      uri = URI.join(uri, path)
+      request = RestClient::Resource.new(uri.to_s, :verify_ssl => OpenSSL::SSL::VERIFY_NONE)
+      begin
+        request.get
+      rescue RestClient::Exception => exception
+        response = exception.response
+        if response.code == 401
+          www_authenticate = response.headers[:www_authenticate]
+          if www_authenticate =~ /^Basic realm="?(RESTAPI|ENGINE)"?$/
+            return true
+          end
+        end
+      end
+      false
+    end
+
+    # Probes all the candidate paths of the API, and returns the first that success. If all probes
+    # fail, then the first candidate will be returned.
+    def find_api_path(uri)
+      CANDIDATE_API_PATHS.detect { |path| probe_api_path(uri, path) } || CANDIDATE_API_PATHS.first
+    end
+
+    # Returns the path of the API, probing it if needed.
+    def api_path
+      @api_path ||= find_api_path(base_uri)
+    end
+
     def api_uri(path = nil)
```

Comment:
@jhernand Can you add some tests for the methods you are changing?  This doesn't work the way that I would expect.

```ruby
irb(main):007:0> c.api_uri
=> "https://dev-rhev35.example.com/api"
irb(main):008:0> c.api_uri("/my/path")
=> "https://dev-rhev35.example.com/api/my/path"
```

I would expect to get `"https://dev-rhev35.example.com/my/path/api"`, wouldn't you?